router/internal/graphqlmetrics/sink.go

```diff
@@ -0,0 +1,23 @@
+package graphqlmetrics
+
+import (
+	"context"
+)
+
+// Sink defines the interface for exporting batches of items to a destination.
+// Implementations must be safe for concurrent use.
+type Sink[T any] interface {
+	// Export sends a batch of items to the sink destination.
+	// It returns an error if the export fails.
+	// The context can be used to cancel or timeout the export operation.
+	Export(ctx context.Context, batch []T) error
+
+	// Close performs any cleanup needed when shutting down the sink.
+	// It should block until all cleanup is complete or the context is cancelled.
+	Close(ctx context.Context) error
+}
```

Comment:
I wonder if this should be put in `pkg` instead? It seems it might be useful as a generic batcher and as far as I can tell doesn't include anything graphqlmetrics specific?